res/config/config_settings.py

```diff
@@ -1,161 +0,0 @@
-#  it under the terms of the GNU General Public License as published by
-#  the Free Software Foundation, either version 3 of the License, or
-#  (at your option) any later version.
-#
-#  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
-#  WARRANTY; without even the implied warranty of MERCHANTABILITY or
-#  FITNESS FOR A PARTICULAR PURPOSE.
-#
-#  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
-#  for more details.
-
-from cwrap import BaseCClass
-
-from res import ResPrototype
-from res.config.content_type_enum import ContentTypeEnum
-
-
-class ConfigSettings(BaseCClass):
-    TYPE_NAME = "config_settings"
-    _alloc = ResPrototype("void* config_settings_alloc(char*)", bind=False)
-    _free = ResPrototype("void  config_settings_free(config_settings)")
-
-    _add_setting = ResPrototype(
-        "bool config_settings_add_setting(config_settings, \
-                                          char*, \
-                                          config_content_type_enum, \
-                                          char*)"
-    )
-    _add_double_setting = ResPrototype(
-        "void config_settings_add_double_setting(config_settings , char* , double)"
-    )
-    _add_int_setting = ResPrototype(
-        "void config_settings_add_int_setting(config_settings , char* , int)"
-    )
-    _add_string_setting = ResPrototype(
-        "void config_settings_add_string_setting(config_settings , char* , char*)"
-    )
-    _add_bool_setting = ResPrototype(
-        "void config_settings_add_bool_setting(config_settings , char* , bool)"
-    )
-    _has_key = ResPrototype("bool config_settings_has_key(config_settings , char*)")
-    _get_type = ResPrototype(
-        "config_content_type_enum config_settings_get_value_type(config_settings, char*)"
-    )
-    _init_parser = ResPrototype(
-        "void config_settings_init_parser(config_settings, config_parser, bool)"
-    )
-    _apply = ResPrototype("void config_settings_apply(config_settings, config_content)")
-    _alloc_keys = ResPrototype(
-        "stringlist_obj config_settings_alloc_keys(config_settings)"
-    )
-
-    _get = ResPrototype("char* config_settings_get_value(config_settings, char*)")
-    _get_int = ResPrototype("int config_settings_get_int_value(config_settings, char*)")
-    _get_double = ResPrototype(
-        "double config_settings_get_double_value(config_settings, char*)"
-    )
-    _get_bool = ResPrototype(
-        "bool config_settings_get_bool_value(config_settings, char*)"
-    )
-
-    _set = ResPrototype("bool config_settings_set_value(config_settings, char*, char*)")
-    _set_int = ResPrototype(
-        "bool config_settings_set_int_value(config_settings, char*, int)"
-    )
-    _set_double = ResPrototype(
-        "bool config_settings_set_double_value(config_settings, char*, double)"
-    )
-    _set_bool = ResPrototype(
-        "bool config_settings_set_bool_value(config_settings, char*, bool)"
```

Comment:
Can you also check whether these functions are used in C++? For example, grepping for this function shows no usages.

Reply:
Yes, I considered doing that, but assumed they must have been used somewhere. I will go ahead and do some grepping!